### PR TITLE
Fix for IADS causing excessive CMDQ entries

### PR DESCRIPTION
**Pull request by wheelyjoe**
Also removed some log prints and minor logic cleanup

### CODE REVIEW

**Comment by wheelyjoe** [comment]
#170

File: src/dct/systems/IADS.lua [hunk 1]
```diff
@@ -508,9 +502,9 @@ function IADS:onShot(event)
 end
 
 function IADS:onBirth(event)
-	if event.initiator.getGroup == nil then
-		return
-	end
+	if event.initiator:getCategory() ~= Object.Category.Unit then
+    return
+  end
```
Comment (jtoppins) [inline src/dct/systems/IADS.lua:516]: The indention looks wrong here.

File: src/dct/systems/IADS.lua [hunk 2]
```diff
@@ -175,12 +167,10 @@ function IADS:enableSAM(site)
 			return
 		end
 	end
-
 	site.group:getController():setOption(
 		AI.Option.Ground.id.ALARM_STATE,
 		AI.Option.Ground.val.ALARM_STATE.RED)
 	site.Enabled = true
-	env.info("SAM: "..site.Name.." enabled")
```
Comment (jtoppins) [inline src/dct/systems/IADS.lua:183]: not sure I understand why we are just removing debug lines like this. Maybe convert to using debug logger messages?

File: src/dct/systems/IADS.lua [hunk 3]
```diff
@@ -175,12 +167,10 @@ function IADS:enableSAM(site)
 			return
 		end
 	end
-
```
Comment (jtoppins) [inline src/dct/systems/IADS.lua:178]: Do not see why we need to remove empty lines like this, reason?

**Review by jtoppins — COMMENTED** [review]
It would be nice to get a commit log message that describes what the problem actually is and how this change fixes it. A simple description of the problem, how the current code causes the problem, and how this change prevents the problem.

**Comment by jtoppins** [comment]
Also you should either reset your master branch with my master branch, and keep it up to date, so that you do not have these merge commits that are not needed. Or setup my tree as a remote and when creating your branch checkout out directly my master branch.

File: src/dct/systems/IADS.lua (hunk 2 above)
Comment (wheelyjoe) [reply src/dct/systems/IADS.lua:183]: They're uneeded and are worse in than out. They don't serve any function for debugging OEO and I can't see why they would elsewhere.

I can readd them later as logger messages, but that's not key to this commit.

File: src/dct/systems/IADS.lua (hunk 3 above)
Comment (wheelyjoe) [reply src/dct/systems/IADS.lua:178]: No reason, looks nicer.

**Comment by wheelyjoe** [comment]
Commit was to stop adding entries to the cmdq as a delaying method and instead have a single scheduled function to check and update status based on a timer in a table.

File: src/dct/systems/IADS.lua [hunk 4]
```diff
@@ -92,12 +92,7 @@ function IADS:__init(cmdr)
 end
 
 function IADS:getSamByName(name)
-  for _, site in pairs(self.SAMSites) do
-    if site.Name == name then
-      return site
-    end
-  end
-  return nil
+	return self.SAMSites(name)
```
Comment (jtoppins) [inline src/dct/systems/IADS.lua:95]: no sure this will work, as far as I know `self.SAMSites` is a table not a function.

**Comment by jtoppins** [comment]
I committed a rebased version of this and added a more descriptive commit log.

**Pull request closed without merging**